### PR TITLE
cli: convert pop to mixin

### PR DESCRIPTION
**Pull request by sh0rez**
Adds initial and highly experimental support for converting pops to
mixins for somewhat easy integration into current pipelines.

Usage:
```
$ pop mix ./path/to/pop [-o yaml|json] [-s alerts|rules|grafana]
```

Depends on #28
Partly implements #22

### CODE REVIEW

File: pkg/pop/mix.cue [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+#dashboardConvert: {
+    #dashboards: [string]: _
+
+    for k, v in #dashboards {
+        "\(k).json": json.Indent(json.Marshal(v), "", "  ")
+    }
+}
+
+#flattenAlerts: {
```
Comment (sdboyer) [inline pkg/pop/mix.cue:39]: why not import the one that's already written and exported?

File: go.sum [hunk 2]
```diff
@@ -0,0 +1,214 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (sdboyer) [inline go.sum:1]: nit: tidy

File: pkg/pop/mix.cue [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import "encoding/json"
+
+// TODO: include full schema?
+// TODO: figure out why it stops working when uncommented (cue eval works tho)
+// prometheusAlerts: v0: {...}
+// prometheusRules: v0: {...}
+// grafanaDashboards: v0: {...}
+
+// TODO: handle different versions, deal with migrations, etc, etc
+mixin: {
```
Comment (sdboyer) [inline pkg/pop/mix.cue:10]: i haven't written a `_test.cue` file yet, but i think they're a thing, and this seems like an appropriate place to start...? :)

**Review by sdboyer — CHANGES_REQUESTED** [review]
Delightfully simple :) Let's get at least one test in, though - the sooner we can start testing, the better

**Comment by sdboyer** [comment]
Oh and also, let's just pretend we've got a comment linter in place - docs on all exported functions, please

File: pkg/pop/mix.cue (hunk 1 above)
Comment (sh0rez) [reply pkg/pop/mix.cue:39]: bad argument: I did not manage to get imports working

good argument: whatever the binary needs should be self contained, we should not rely on the users cue.mod to have what we need. that also makes it predictable

File: pkg/pop/mix.cue (hunk 3 above)
Comment (sdboyer) [reply pkg/pop/mix.cue:10]: derp. seems like i just totally imagined that.

well. `_test.go` files it is, then :D

File: pkg/pop/mix.cue (hunk 1 above)
Comment (sdboyer) [reply pkg/pop/mix.cue:39]: Generally agreed, but i think that for this case, it's fine to embed everything contained in the same CUE module and thereby treat those imports as safe. By CUE's own rules, those don't trigger package management.

In fact, i'd even argue that not including the entire tree that shares an import namespace/`cue.mod` actually violates the principle of least surprise inasmuch as it disallows same-module imports.

File: go.sum (hunk 2 above)
Comment (sdboyer) [reply go.sum:1]: oh wait, these must all be transitive dependencies of cue then. OK

**Comment by talentedmrjones** [comment]
@sh0rez just curious, what had you choose `go-clix/cli` over `spf13/cobra`? While go-clix/cli is based on cobra its stats are concerning: 10 stars, last commit 7 months ago, vs 22k stars last commit yesterday. Also for whatever it's worth, `cue` and some other tooling I'm seeing in the community are all using cobra. If you have specific reasons for continuing with go-clix I would love to hear them otherwise I would vote to switch to cobra before more of us start contributing. Thanks for considering my thoughts!

File: cmd/pop/eval.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+		if err != nil {
+			return err
+		}
+
+		mix, err := p.Mixin()
+		if err != nil {
+			return err
+		}
+
+		var out interface{} = mix
```
Comment (metalmatze) [inline cmd/pop/eval.go:29]: You could inline choosePrinter and not have to use `interface{}` anymore, if I'm not mistaken.

**Comment by sh0rez** [comment]
@talentedmrjones:


> @sh0rez just curious, what had you choose go-clix/cli over spf13/cobra? While go-clix/cli is based on cobra its stats are concerning: 10 stars, last commit 7 months ago, vs 22k stars last commit yesterday. 

`go-clix/cli` was created by me and my team at Grafana Labs while working on [Tanka](github.com/grafana/tanka), mostly as an experiment whether we could reduce the dependency surface of our cli library (which indeed worked out). We also performed certain modifications to the API we considered a convenience of which built-in argument parsing and suggesting is the biggest one (we can suggest CLI completions directly from Go).

> Also for whatever it's worth, cue and some other tooling I'm seeing in the community are all using cobra. If you have specific reasons for continuing with go-clix I would love to hear them otherwise I would vote to switch to cobra before more of us start contributing. Thanks for considering my thoughts!

That's a fair point, looking at the ecosystem is important! I wouldn't be too worried personally, given we can (1) change go-clix at will and (2) the API is very similar, so a migration at a later point likely won't hurt us much. 

What do you think?

**Review by sdboyer — APPROVED** [review]
Sorry this took so long! Looks good enough to work from

**Pull request merged**